dotnet/src/SemanticKernel.UnitTests/AI/FunctionChoiceBehaviors/RequiredFunctionChoiceBehaviorTests.cs

```diff
@@ -0,0 +1,292 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.SemanticKernel;
+using Xunit;
+
+namespace SemanticKernel.UnitTests.AI.FunctionChoiceBehaviors;
+
+/// <summary>
+/// Unit tests for <see cref="RequiredFunctionChoiceBehavior"/>
+/// </summary>
+public sealed class RequiredFunctionChoiceBehaviorTests
+{
+    private readonly Kernel _kernel;
+
+    public RequiredFunctionChoiceBehaviorTests()
+    {
+        this._kernel = new Kernel();
+    }
+
+    [Fact]
+    public void ItShouldAdvertiseAllKernelFunctions()
+    {
+        // Arrange
+        var plugin = GetTestPlugin();
+        this._kernel.Plugins.Add(plugin);
+
+        // Act
+        var choiceBehavior = new RequiredFunctionChoiceBehavior();
+
+        var config = choiceBehavior.GetConfiguration(new([]) { Kernel = this._kernel });
+
+        // Assert
+        Assert.NotNull(config);
+
+        Assert.NotNull(config.Functions);
+        Assert.Equal(3, config.Functions.Count);
+        Assert.Contains(config.Functions, f => f.Name == "Function1");
+        Assert.Contains(config.Functions, f => f.Name == "Function2");
+        Assert.Contains(config.Functions, f => f.Name == "Function3");
+    }
+
+    [Fact]
+    public void ItShouldAdvertiseOnlyFunctionsSuppliedViaConstructor()
+    {
+        // Arrange
+        var plugin = GetTestPlugin();
+        this._kernel.Plugins.Add(plugin);
+
+        // Act
+        var choiceBehavior = new RequiredFunctionChoiceBehavior(functions: [plugin.ElementAt(0), plugin.ElementAt(1)]);
+
+        var config = choiceBehavior.GetConfiguration(new([]) { Kernel = this._kernel });
+
+        // Assert
+        Assert.NotNull(config);
+
+        Assert.NotNull(config.Functions);
+        Assert.Equal(2, config.Functions.Count);
+        Assert.Contains(config.Functions, f => f.Name == "Function1");
+        Assert.Contains(config.Functions, f => f.Name == "Function2");
+    }
+
+    //[Fact]
```

Comment:
Should this still be commented out?

Reply:
Not yet, but it will be done in one of the follow-up PRs for supporting the new function-calling model in JSON/YAML prompts.